addons/mail/static/src/models/user_notification_manager.js

```diff
@@ -0,0 +1,142 @@
+/** @odoo-module **/
+
+import { registerModel } from '@mail/model/model_core';
+import { attr } from '@mail/model/model_field';
+
+import { url } from '@web/core/utils/urls';
+
+registerModel({
+    name: 'UserNotificationManager',
+    identifyingFields: [],
+    recordMethods: {
+        /**
+         * Send a notification, preferably a native one. If native
+         * notifications are disable or unavailable on the current
+         * platform, fallback on the notification service.
+         *
+         * @param {Object} param0
+         * @param {string} [param0.message] The body of the
+         * notification.
+         * @param {string} [param0.title] The title of the notification.
+         * @param {string} [param0.type] The type to be passed to the no
+         * service when native notifications can't be sent.
+         * @returns
+         */
+        sendNotification({ message, title, type }) {
+            if (!this.messaging.env.services['multi_tab'].isOnMainTab()) {
+                return;
+            }
+            if (this.canSendNativeNotification) {
+                try {
+                    this._sendNativeNotification(title, message);
+                } catch (error) {
+                    // Notification without Serviceworker in Chrome Android doesn't works anymore
+                    // So we fallback to the notification service in this case
+                    // https://bugs.chromium.org/p/chromium/issues/detail?id=481856
+                    if (error.message.includes('ServiceWorkerRegistration')) {
+                        this._sendOdooNotification(message, { title, type });
+                    } else {
+                        throw error;
+                    }
+                }
+            } else {
+                this._sendOdooNotification(message, { title, type });
+            }
+        },
+        /**
+         * @private
+         * @returns {HTMLAudioElement}
+         */
+        _computeAudio() {
+            const audioElement = new Audio();
+            const ext = audioElement.canPlayType("audio/ogg; codecs=vorbis") ? ".ogg" : ".mp3";
+            audioElement.src = url("/mail/static/src/audio/ting" + ext);
```

Comment:
I would prefer if the full file name was build twice, much easier to find it in code than using string manipulation.